### PR TITLE
add a source debug build

### PR DESCRIPTION
**Pull request by mikebrow**
Adds `make debug`.. allows a developer to build a source debuggable version of skopeo. For example, using delve.

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

**Comment by runcom** [comment]
@duglin ptal

File: Makefile [hunk 1]
```diff
@@ -41,6 +41,20 @@ skopeo: cmd/skopeo
 binary-local:
 	go build -ldflags "-X main.gitCommit=${GIT_COMMIT}" -o skopeo ./cmd/skopeo
 
+#   make debug
+#     Note: Uses the -N -l go compiler options to disable compiler optimizations
+#           and inlining. Using these build options allows you to subsequently
+#           use source debugging tools like delve.
+debug: debug-skopeo docs
```
Comment (mtrmac) [inline Makefile]: Is this useful? I would expect that when debugging, you only need the binary.

File: Makefile [hunk 2]
```diff
@@ -41,6 +41,20 @@ skopeo: cmd/skopeo
...
+
+debug-skopeo: cmd/skopeo
+	docker build ${DOCKER_BUILD_ARGS} -f Dockerfile.build -t skopeobuildimage .
+	docker run --rm -v ${PWD}:/src/github.com/projectatomic/skopeo \
+		skopeobuildimage make debug-binary-local
+
+debug-binary-local:
+	go build -ldflags "-X main.gitCommit=${GIT_COMMIT}" -gcflags $(GOGCFLAGS) -o skopeo ./cmd/skopeo
+
```
Comment (mtrmac) [inline Makefile]: All of this duplication, and top-level targets, feels a bit unclean; would it be reasonably easy to have `make {binary,binary-local} DEBUG=1` or something like that, and use the DEBUG flag to decide whether to set the `-gcflags`?

File: Makefile (hunk 1 above)
Comment (mikebrow) [inline Makefile]: make install currently requires the doc be built first .. otherwise yeah sure we don't have to build docs

Comment (mtrmac) [inline Makefile]: The just added #163 could be another way to deal with that.

File: Makefile (hunk 2 above)
Comment (mikebrow) [inline Makefile]: I suppose..

**Comment by mikebrow** [comment]
@mtrmac Switched to a make ... DEBUG=1 pattern

File: Makefile [hunk 3]
```diff
@@ -35,11 +47,12 @@ binary: skopeo
 skopeo: cmd/skopeo
 	docker build ${DOCKER_BUILD_ARGS} -f Dockerfile.build -t skopeobuildimage .
 	docker run --rm -v ${PWD}:/src/github.com/projectatomic/skopeo \
-		skopeobuildimage make binary-local
+		skopeobuildimage make binary-local ${DEBUGENV}
```
Comment (mtrmac) [inline Makefile]: Can just do `DEBUG=$(DEBUG)` here without the extra `DEBUGENV` variable.

Comment (mikebrow) [inline Makefile]: Works, adds the DEBUG=0 to the output of the make all which I suppose is ok.

**Comment by mikebrow** [comment]
rebased...

**Comment by mtrmac** [comment]
Merged, thanks!

**Comment by mtrmac** [comment]
Bah, I can’t merge this manually it seems.

@mikebrow do you want to try rebasing again?  Sorry about that… Or we can wait a few days for @runcom or somebody else with admin status on this project to allow force-pushing.

**Comment by mikebrow** [comment]
Saw the merge and stopped watching for additional messages, sry bout that.  Rebased.

**Pull request merged**